pinot-controller/pom.xml

```diff
@@ -121,13 +122,13 @@
             </configuration>
           </execution>
           <execution>
-            <id>npm ci</id>
+            <id>npm ci</id><!-- npm ci removes node_modules! -->
```

Comment:
IIRC npm install actually modifies `package.json` in order to install new versions of the libraries, which is something we don't want.

Reply:
You're right. Let me check if we can get an even better speedup (and matching behavior) by switching to another package manager (current one is horribly slow).

Reply:
I reverted the change and added up-to-date check between source js files and main.js, executed only for pinot-fastdev profile.